src/cpp/polars/TimeSeries.h

```diff
@@ -0,0 +1,143 @@
+//
+// Created by Linda Uruchurtu on 03/10/2018.
+//
+
+#ifndef POLARS_TIMESERIES_H
+#define POLARS_TIMESERIES_H
+
+#include <cassert>
+#include <vector>
+#include <cmath>
+#include <string>
+#include "armadillo"
+#include <chrono>
+#include "date/date.h"
+
+#include "Series.h"
+
+
+namespace polars {
+
+    using namespace std::chrono;
+    typedef std::chrono::duration<double> unix_epoch_seconds;
+
+    template<class TimePointType>
+    class TimeSeries : public Series {
+
+    public:
+        using Series::loc;
+
+        TimeSeries() = default;
+
+        TimeSeries(arma::vec v0, std::vector<TimePointType> t0){
+            v = v0;
+            t = chrono_to_double_vector(t0);
+        };
+
+        std::vector<TimePointType> timestamps() const {
+            // Pass indices and return vector of timepoints
+            return double_to_chrono_vector(index());
+        };
+
+        std::map<TimePointType, double> to_timeseries_map() const {
+
+            std::map<TimePointType, double> m;
+
+            std::vector<TimePointType> timepoints = double_to_chrono_vector(index());
+
+            // Put pairs into map
+            for(int i = 0; i < size(); i++) {
+                m.insert(std::make_pair(timepoints[i], values()[i]));
+            }
+
+            return m;
+        };
```

Comment:
Based on how templated hide base methods, we might be able to call this `.to_map()` if we want to

Reply:
added todo